dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/ManifestInfo.scala

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.dev
+
+import java.io.IOException
+import java.util
+import java.util.Arrays
+import java.util.jar.{ Attributes, Manifest }
+
+import play.dev.filewatch.LoggerProxy
+
+import scala.collection.immutable
+
+// TODO: if we move it out to another project, we should reconsider how to do logging
+final class ManifestInfo(val classLoader: ClassLoader, logger: LoggerProxy) {
+  import ManifestInfo._
+
+  /**
+   * Versions of artifacts from known vendors.
+   */
+  val versions: Map[String, Version] = {
+
+    var manifests = Map.empty[String, Version]
+
+    try {
+      val resources = classLoader.getResources("META-INF/MANIFEST.MF")
+      while (resources.hasMoreElements) {
+        val ios = resources.nextElement().openStream()
+        try {
+          val manifest = new Manifest(ios)
+          val attributes = manifest.getMainAttributes
+          val title = attributes.getValue(new Attributes.Name(ImplTitle)) match {
+            case null ⇒ attributes.getValue(new Attributes.Name(BundleName))
+            case t    ⇒ t
+          }
+          val version = attributes.getValue(new Attributes.Name(ImplVersion)) match {
+            case null ⇒ attributes.getValue(new Attributes.Name(BundleVersion))
+            case v    ⇒ v
+          }
+          val vendor = attributes.getValue(new Attributes.Name(ImplVendor)) match {
+            case null ⇒ attributes.getValue(new Attributes.Name(BundleVendor))
+            case v    ⇒ v
+          }
+
+          if (title != null
+            && version != null
+            && vendor != null
+            && knownVendors(vendor)) {
```

Comment:
this is something we should reconsider if we decide to make it an apart lig/plugin